server/src/main/java/org/opensearch/snapshots/RestoreService.java

```diff
@@ -486,8 +486,18 @@ public ClusterState execute(ClusterState currentState) {
                                         // Remove all aliases - they shouldn't be restored
                                         indexMdBuilder.removeAllAliases();
                                     } else {
-                                        for (final String alias : snapshotIndexMetadata.getAliases().keySet()) {
-                                            aliases.add(alias);
+                                        for (final Map.Entry<String, AliasMetadata> alias : snapshotIndexMetadata.getAliases().entrySet()) {
+                                            String aliasName = alias.getKey();
+                                            if (request.renameAliasPattern() != null && request.renameAliasReplacement() != null) {
```

Comment:
This null check can be done outside the loop

Reply:
Okay, done.